### DISCUSSION TITLE
1ms processing time?

### DISCUSSION THREAD

**Discussion by siewers** [AdguardTeam/AdGuardHome#4307] [Q&A]
1ms in average processing time seem unrealistic to me, but I'm not complaining. I just wonder if there's something wrong in the measurement or somehow another bug on my end?

I'm on v0.107.3 and use the following upstream DNS servers (parallel requests):
- 208.67.220.220
- 208.67.222.222
- 1.1.1.1
- 1.0.0.1
- quic://dns-unfiltered.adguard.com:784

I'm only using the following blocklists:
- OISD adblocker-syntax domains (full) (https://abp.oisd.nl)
- AdGuard DNS filter (https://adguardteam.github.io/AdGuardSDNSFilter/Filters/filter.txt)

The hardware is a Raspberry Pi 4 Model B 4GB

![image](https://user-images.githubusercontent.com/226147/154558811-00b5cb9d-5720-4d8c-8816-04011090f166.png)

**Comment by ainar-g** [comment]
Hello.  The number does look odd, but not completely improbable.  UDP queries can be performed quite quickly, on the order of hundreds of microseconds.  But the thing that worries me is that you still use filtering, which on a Pi 4 should probably take a bit longer than that.  Do you see the same order of durations in the query log?  Also, are you sure that the “Block domains using filters and hosts files” checkbox is checked on the “General settings” page?

**Reply by siewers** [reply]
Yes, it's enabled and it's blocking about 15% of the requests.

![image](https://user-images.githubusercontent.com/226147/154701915-f9b49cbe-737f-417a-a3f1-5e9a1d4660da.png)

**Reply by ainar-g** [reply]
Sorry about the late response.  If there are no errors, and the server works properly, this shouldn't be an issue.

**Reply by siewers** [reply]
That's great, but isn't it a bit unusual? I mean, it seems unlikely and leads me to believe the numbers are off, so I'm curious if this is common?
Don't get me wrong, I am not complaining, I'm just wondering if something is off in my setup somehow.

**Reply by ainar-g** [reply]
That depends largely on your upstreams.  Perhaps it's somewhere close on the network, or even on the same machine?
